stl/inc/ranges

```diff
@@ -4748,6 +4748,378 @@ namespace ranges {
+                const _Iterator& _Right) noexcept(noexcept(_Left._Current - _Right._Current)) /* strengthened */
+                requires sized_sentinel_for<_Base_iter, _Base_iter> {
+                if constexpr (_Slide_caches_first<_Base>) {
+                    return _Left._Last_element - _Right._Last_element;
+                } else {
+                    return _Left._Current - _Right._Current;
+                }
+            }
+        };
+
+        class _Sentinel {
+        private:
+            friend slide_view;
+            /* [[no_unique_address]] */ sentinel_t<_Vw> _Last{};
+
+            constexpr explicit _Sentinel(sentinel_t<_Vw> _Last_) noexcept(
+                is_nothrow_move_constructible_v<sentinel_t<_Vw>>) /* strengthened */
+                : _Last(_STD move(_Last_)) {}
+
+        public:
+            _Sentinel() = default;
+
+            _NODISCARD_FRIEND constexpr bool
+                operator==(const _Iterator<false>& _Left, const _Sentinel& _Right) noexcept(noexcept(
+                    _Implicitly_convert_to<bool>(_Left._Get_last_element() == _Right._Last))) /* strengthened */ {
+                return _Left._Get_last_element() == _Right._Last;
+            }
+
+            _NODISCARD_FRIEND constexpr range_difference_t<_Vw> operator-(const _Iterator<false>& _Left,
+                const _Sentinel& _Right) noexcept(noexcept(_Left._Get_last_element() - _Right._Last)) /* strengthened */
+                requires sized_sentinel_for<sentinel_t<_Vw>, iterator_t<_Vw>> {
+                return _Left._Get_last_element() - _Right._Last;
+            }
+
+            _NODISCARD_FRIEND constexpr range_difference_t<_Vw>
+                operator-(const _Sentinel& _Left, const _Iterator<false>& _Right) noexcept(
+                    noexcept(_Left._Last - _Right._Get_last_element())) /* strengthened */
+                requires sized_sentinel_for<sentinel_t<_Vw>, iterator_t<_Vw>> {
+                return _Left._Last - _Right._Get_last_element();
+            }
+        };
+
+    public:
+        // clang-format off
+        slide_view() requires default_initializable<_Vw> = default;
+        // clang-format on
+
+        constexpr explicit slide_view(_Vw _Range_, const range_difference_t<_Vw> _Count_) noexcept(
+            is_nothrow_move_constructible_v<_Vw>) /* strengthened */
+            : _Range(_STD move(_Range_)), _Count{_Count_} {}
+
+        // clang-format off
+        _NODISCARD constexpr auto begin() requires (!(_Simple_view<_Vw> && _Slide_caches_nothing<const _Vw>) ) {
+            // clang-format on
+            if constexpr (_Slide_caches_first<_Vw>) {
+                auto _First = _RANGES begin(_Range);
+                if (this->_Has_cache()) {
+                    return _Iterator<false>{_First, this->_Get_cache(_Range), _Count};
+                }
+
+                auto _Result = _RANGES next(_First, _Count - 1, _RANGES end(_Range));

```

Comment:
nit: I do not love the name `_Result` here; I'd prefer something like `_Last` or `_Current_last` or something.